codegen/src/generator/rest_json.rs

```diff
@@ -115,6 +115,78 @@ impl GenerateProtocol for RestJsonGenerator {
     }
 }
 
+fn http_code_to_status_code(code: Option<i32>) -> String {
+    match code {
+        Some(actual_code) => {
+            // From http://hyper.rs/hyper/v0.9.12/src/hyper/src/status.rs.html and modified to return strings:
```

Comment:
Instead of overwriting the trait implementation for `StatusCode` how about we make a new one?  [Here's a playground example](https://is.gd/Wv3ReB) of what I'm thinking.

I suggest this because I'm not sure if our overriding of it in codegen will prevent us from using the more-human-readable-format later for surfacing error messages to crate users.